### PR TITLE
8277165: jdeps --multi-release --print-module-deps fails if module-info.class in different versioned directories

### PR DESCRIPTION
**Pull request by mlchung**
jdeps intends to report an error if there are multiple versions of the same class being parsed.   module-info.class should be excluded from such detection.

This patch also fixes a data race in `VersionHelper::set` and also unwraps the `ExecutionException` when FutureTask of parsing the classes throws an exception to report `MultiReleaseException` properly.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issues
 * [JDK-8277165](https://bugs.openjdk.java.net/browse/JDK-8277165): jdeps --multi-release --print-module-deps fails if module-info.class in different versioned directories
 * [JDK-8277166](https://bugs.openjdk.java.net/browse/JDK-8277166): Data race in jdeps VersionHelper
 * [JDK-8277123](https://bugs.openjdk.java.net/browse/JDK-8277123): jdeps does not report some exceptions correctly

### Reviewers
 * [Jorn Vernee](https://openjdk.java.net/census#jvernee) (@JornVernee - **Reviewer**) ⚠️ Review applies to 2cb770d4ec24426d8139c221d3fe0600a1dd29e8
 * [Alan Bateman](https://openjdk.java.net/census#alanb) (@AlanBateman - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk pull/6530/head:pull/6530` \
`$ git checkout pull/6530`

Update a local copy of the PR: \
`$ git checkout pull/6530` \
`$ git pull https://git.openjdk.java.net/jdk pull/6530/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 6530`

View PR using the GUI difftool: \
`$ git pr show -t 6530`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk/pull/6530.diff">https://git.openjdk.java.net/jdk/pull/6530.diff</a>

</details>

### CODE REVIEW

**Review by JornVernee — APPROVED** [review]
LGTM. Thanks for fixing.

File: src/jdk.jdeps/share/classes/com/sun/tools/jdeps/DependencyFinder.java [hunk 1]
```diff
@@ -268,7 +268,14 @@ boolean isParsed(Location location) {
...
             return targets;
         } catch (InterruptedException|ExecutionException e) {
-            throw new Error(e);
+            Throwable t = e.getCause();
+            if (t instanceof RuntimeException) {
+                throw (RuntimeException)t;
+            } else if (t instanceof Error) {
+                throw (Error)t;
+            } else {
+                throw new Error(e);
```
Comment (AlanBateman) [inline src/jdk.jdeps/share/classes/com/sun/tools/jdeps/DependencyFinder.java:277]: A minor suggestion is that you could avoid the casts with:

Throwable cause = ...
if (cause instanceof RuntimeException e) {
    throw e;
} else if (cause instanceof Error e) {
    throw e;
} else {

**Review by AlanBateman — APPROVED** [review]

File: src/jdk.jdeps/share/classes/com/sun/tools/jdeps/DependencyFinder.java (hunk 1 above)
Comment (mlchung) [reply src/jdk.jdeps/share/classes/com/sun/tools/jdeps/DependencyFinder.java:277]: Fixed.  Thanks for reminding.

**Review by AlanBateman — APPROVED** [review]

**Comment by jaikiran** [comment]
Hello Mandy, one small issue - `ClassFileReader`, `DependencyFinder` and `VersionHelper` are misisng a copyright year update for this change.

**Comment by mlchung** [comment]
@jaikiran Sorry, I missed your comment about the copyright year update.  I will update them in other jdeps fix.

**Pull request closed without merging**